components/tools/OmeroPy/src/omero/plugins/prefs.py

```diff
@@ -482,11 +482,18 @@ def edit(self, args, config, edit_path=edit_path):
             self.ctx.die(954, "%s: Failed to edit %s"
                          % (getattr(re, "pid", "Unknown"), temp_file))
         args.NAME = config.default()
+        old_config = dict(config)
         self.drop(args, config)
         args.file = [open(str(temp_file), "r")]
         args.q = True
-        self.load(args, config)
-        remove_path(temp_file)
+        try:
+            self.load(args, config)
+        except Exception as e:
+            for key, value in old_config.items():
```

Comment:
This could be: `config.update(old_config)`:

```
>>> a = {1:2}
>>> b = {1:10, 3:4}
>>> a.update(b)
>>> a
{1: 10, 3: 4}
```

Reply:
`AttributeError: 'ConfigXml' object has no attribute 'update'`

Reply:
My apologies. That would have had to been on the underlying dictionary.

Reply:
np, this `config` thing is tricksy!